engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -168,8 +171,10 @@ class WorkflowManagerActor(backend: Backend, config: Config)
 
   when (Running) {
     case Event(SubmitWorkflow(source), _) =>
-      val updatedData = submitWorkflow(source, replyTo = Option(sender), id = None)
-      stay() using updatedData
+      submitWorkflow(source, replyTo = Option(sender), id = None) onSuccess {
+        case updatedData: WorkflowManagerData => self ! UpdateWorkflowManagerState(updatedData)
+      }
+      stay()
```

Comment:
This should not be passing data to set later.  There could be messages that come in before this one arrives that change the data in unrelated ways, but the updates from the processing of those messages would be overwritten.